testnet/stacks-node/src/neon_node.rs

```diff
@@ -401,7 +402,8 @@ fn spawn_miner_relayer(
     let mut mem_pool = MemPoolDB::open(false, TESTNET_CHAIN_ID, &stacks_chainstate_path)
         .map_err(NetError::DBError)?;
 
-    let mut last_mined_blocks = vec![];
+    let mut last_mined_blocks: HashMap<BurnchainHeaderHash, Vec<AssembledAnchorBlock>> =
```

Comment:
Trying to understand this change.  We need to keep `last_mined_blocks` grouped by burnchain header hash because the burnchain can fork, and the miner can be working on two or more burnchain forks at once?

Reply:
Yes -- this deals with the following situation (which occurs in the regtest forking integration test, but would probably be rare in practice):

A -> B -> C -> D
          \_> C' -> D' -> E'

1. Miner mines on burnchain block "C", so the block commit would be included in "D".
2. Before D is mined, C'->D'->E' is discovered by the miner, so the current chain tip is E'
3. `last_mined_blocks` would contain block commits on a different chain tip -- as previously implemented, the miner would deduce that an RBF should be attempted, and try to re-spend a UTXO that was forked out of existence.

With a burn header hash-keyed map, this problem goes away, because the miner would only see their prior attempts at the same burnchain tip. If the C->D chain regained power and became part of the canonical chain again, the miner would have their corresponding block commit data that could be broadcasted if they won the sortition.